backend/pkg/modules/internal/bfla/bfladetector/user_detection.go

```diff
@@ -0,0 +1,70 @@
+// Copyright © 2022 Cisco Systems, Inc. and its affiliates.
+// All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package bfladetector
+
+import (
+	"encoding/base64"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"strings"
+)
+
+func GetUserID(headers map[string]string) (*DetectedUser, error) {
+	if xcustomerID, ok := headers["x-customer-id"]; ok {
+		return &DetectedUser{Source: DetectedUserSourceXConsumerIDHeader, ID: xcustomerID}, nil
+	}
+	authz, ok := headers["authorization"]
```

Comment:
GetUserID is given a headers map where keys can be mixed case.
I case of a `AuThorIzation` header name, which is valid, you won't be have to get the userID.